src/common/snippets/src/lowered/pass/compute_buffer_allocation_size.cpp

```diff
@@ -0,0 +1,100 @@
+
+    const auto& subtensor =  parent_port.get_descriptor_ptr()->get_subtensor();
+
+    size_t allocation_size = 1;
+    std::set<size_t> processed_dim_idxs;
+    for (const auto& parent_loop : parent_loop_ids) {
+        const auto loop_info = loop_manager->get_loop_info(parent_loop);
+        const auto& output_ports = loop_info->get_output_ports();
+        auto it = std::find_if(output_ports.begin(), output_ports.end(), [&parent_port](const LoopPort& port) { return *port.expr_port == parent_port; });
+        OPENVINO_ASSERT(it != output_ports.end(), "compute_allocation_shape: output port of parent loop can not be found");
+        const auto& loop_port = *it;
+        const auto& dim_idx = loop_port.dim_idx;
+        if (loop_port.is_incremented && dim_idx < rank) {
+            if (const auto& unified_loop_info = ov::as_type_ptr<UnifiedLoopInfo>(loop_info))
+                utils::dynamic_safe_mul(allocation_size, unified_loop_info->get_work_amount());
+            else if (const auto& expanded_loop_info = ov::as_type_ptr<ExpandedLoopInfo>(loop_info))
+                utils::dynamic_safe_mul(allocation_size, expanded_loop_info->get_unified_loop_info()->get_work_amount());
+            else
+                OPENVINO_THROW("Unknown LoopInfo type");
+            processed_dim_idxs.insert(dim_idx);
+        }
+    }
+    const auto processing_rank = !processed_dim_idxs.empty() ? std::max(*processed_dim_idxs.rbegin(), subtensor.size()) : subtensor.size();
+    for (size_t i = 0; i < std::min(processing_rank, rank); ++i) {
+        if (processed_dim_idxs.count(i) == 0) {
+            if (i < subtensor.size())
+                utils::dynamic_safe_mul(allocation_size, std::min(*(planar_shape.rbegin() + i), *(subtensor.rbegin() + i)));
+            else
+                utils::dynamic_safe_mul(allocation_size, *(planar_shape.rbegin() + i));
+        }
+    }
+
+    // Corner case when the current information is not enough
+    if (processing_rank == 0 && processed_dim_idxs.empty()) {
+        for (size_t i = 0; i < rank; ++i) {
+            utils::dynamic_safe_mul(allocation_size, *(planar_shape.rbegin() + i));
+        }
+    }
+
+    return allocation_size;
+}
+
+bool ComputeBufferAllocationSize::run(LinearIR& linear_ir, lowered::LinearIR::constExprIt begin, lowered::LinearIR::constExprIt end) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::ComputeBufferAllocationSize")
+
+    const auto& loop_manager = linear_ir.get_loop_manager();
+
+    const auto& buffer_expressions = linear_ir.get_buffers();
+    for (const auto& buffer_expr : buffer_expressions) {
+        const auto node = buffer_expr->get_node();
+        OPENVINO_ASSERT(ov::is_type<op::Buffer>(node), "Expected Buffer ops in Buffer expressions of LinearIR");
+        if (const auto buffer = ov::as_type_ptr<op::IntermediateMemoryBuffer>(node)) {
+            // If the current size is undefined, update it
+            if (!buffer->is_defined())
```

Comment:
This condition makes the transformation not-reuseble from runtime configurator.
It seems that it would be much easier to differentiate between static and dynamic buffers here and to update dynamic ones every time the transformation is called.
Ideally, configurator should not care about each and every transformation and call anything directly. It should simply run a bunch of optimized passes and propagate their results to runtime info/args.

Reply:
Discussed offline: there should be different Buffer classes for that (as for LoopInfo), for example. Probably the specific Buffer Expressions. Should be separately analyzed in bounds of CVS-143395